test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -74,17 +79,17 @@ var _ = Describe("Ensure OwnerReferences and Finalizers are resilient with Failu
 				Flavor:                ptr.To(testSpecificSettingsGetter().FlavorForMode("ownerrefs-finalizers")),
 				PostNamespaceCreated:  testSpecificSettingsGetter().PostNamespaceCreatedFunc,
 				PostMachinesProvisioned: func(proxy framework.ClusterProxy, namespace, clusterName string) {
-					// check the cluster identity secret has expected ownerReferences and finalizers, and they are resilient
-					// Note: identity secret is not part of the object graph, so it requires an ad-hoc test.
-					checkClusterIdentitySecretOwnerRefAndFinalizer(ctx, proxy.GetClient())
+					if testMode == GovmomiTestMode {
+						// check the cluster identity secret has expected ownerReferences and finalizers, and they are resilient
+						// Note: identity secret is not part of the object graph, so it requires an ad-hoc test.
+						checkClusterIdentitySecretOwnerRefAndFinalizer(ctx, proxy.GetClient())
+					}
 
-					// Set up a periodic patch to ensure the DeploymentZone anc ClusterResourceSets are reconciled.
+					// Set up a periodic patch to ensure the DeploymentZone are reconciled.
 					// Note: this is required because DeploymentZone are not watching for clusters, and thus the DeploymentZone controller
 					// won't be triggered when we un-pause clusters after modifying objects ownerReferences & Finalizers to test resilience.
-					// WRT to ClusterResourceSets, we are forcing reconcile to avoid the issue described in https://github.com/kubernetes-sigs/cluster-api/pull/10656;
-					// we should reconsider if possible to drop force reconcile after this PR is merged.
 					forceCtx, forceCancelFunc := context.WithCancel(ctx)
-					forcePeriodicReconcile(forceCtx, proxy.GetClient(), namespace)
+					forcePeriodicReconcile(forceCtx, proxy.GetClient())
```

Comment:
Can we move this into the if branch from l.82? VSphereDeploymentZones don't exist in supervisor

(I know we have to keep l.91 outside of the if so we can cancel below, but I think it's still worth it not having that running in supervisor tests)